leader-election-impl/src/main/java/com/palantir/paxos/SplittingPaxosStateLog.java

```diff
@@ -62,12 +67,47 @@ private SplittingPaxosStateLog(PaxosStateLog<V> legacyLog,
         return new SplittingPaxosStateLog<>(
                 parameters.legacyLog(),
                 parameters.currentLog(),
-                parameters.markLegacyWrite(),
-                parameters.markLegacyRead(),
+                parameters.legacyOperationMarkers().markLegacyWrite(),
+                parameters.legacyOperationMarkers().markLegacyRead(),
                 parameters.cutoffInclusive(),
                 new AtomicLong(parameters.legacyLog().getLeastLogEntry()));
     }
 
+    public static <V extends Persistable & Versionable> PaxosStateLog<V> createWithMigration(
+            PaxosStorageParameters params,
+            Persistable.Hydrator<V> hydrator,
+            LegacyOperationMarkers legacyOperationMarkers,
+            OptionalLong migrateFrom) {
+        String logDirectory = params.fileBasedLogDirectory()
+                .orElseThrow(() -> new SafeIllegalStateException("We currently need to have file-based storage"));
+        NamespaceAndUseCase namespaceUseCase = params.namespaceAndUseCase();
+
+        PaxosStateLogMigrator.MigrationContext<V> migrationContext = ImmutableMigrationContext.<V>builder()
+                .sourceLog(new PaxosStateLogImpl<>(logDirectory))
```

Comment:
This is fine. I would suggest we replace this with a static factory called `createFileBacked(...)` or something like that for readability

Reply:
non-blocking